File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        tempnums = []
        
        if len(nums) != 0:
            tempnums.append(nums[0])

        for i in range(1,len(nums)):
            if nums[i] in tempnums:
                return True
            else:
                tempnums.append(nums[i])
                continue
        
        return False
            

File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numMap;

        for(int i = 0; i < nums.size(); i++){
            numMap[nums[i]] = i;
        }

        for(int i = 0; i < nums.size(); i++){
            int ans = target - nums[i];
            if(numMap.count(ans) && numMap[ans] != i){
                return {i, numMap[ans]};
            }
        }
        return {};
    }
};
